snuba/datasets/sessions.py

```diff
@@ -0,0 +1,270 @@
+                (
+                    "duration",
+                    AggregateFunction("quantilesIf(0.5, 0.9)", UInt(32), UInt(8)),
+                ),
+                ("uniq_sessions", AggregateFunction("countIf", UUID(), UInt(8))),
+                ("uniq_users", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_sessions_crashed", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                (
+                    "uniq_sessions_abnormal",
+                    AggregateFunction("uniqIf", UUID(), UInt(8)),
+                ),
+                ("uniq_sessions_errored", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_users_crashed", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_users_abnormal", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_users_errored", AggregateFunction("uniqIf", UUID(), UInt(8))),
+            ]
+        )
+        read_schema = AggregatingMergeTreeSchema(
+            columns=read_columns,
+            local_table_name=READ_LOCAL_TABLE_NAME,
+            dist_table_name=READ_DIST_TABLE_NAME,
+            prewhere_candidates=["project_id", "org_id"],
+            order_by="(org_id, project_id, started, release)",
+            partition_by="(toMonday(started))",
+            settings={"index_granularity": 16384},
+        )
+        materialized_view = MaterializedViewSchema(
+            local_materialized_view_name=READ_LOCAL_MV_NAME,
+            dist_materialized_view_name=READ_DIST_MV_NAME,
+            prewhere_candidates=["project_id", "org_id"],
+            columns=read_columns,
+            query="""
+                SELECT
+                    org_id,
+                    project_id,
+                    toStartOfHour(started) as started,
+                    release,
+                    environment,
+                    quantilesIfState(0.5, 0.9)(
+                        duration,
+                        duration <> 4294967295 AND status == 1
+                    ) as duration,
+                    countIfState(session_id, seq == 0) as uniq_sessions,
+                    uniqIfState(distinct_id, distinct_id != '00000000-0000-0000-0000-000000000000') as uniq_users,
+                    uniqIfState(session_id, status == 2) as uniq_sessions_crashed,
+                    uniqIfState(session_id, status == 3) as uniq_sessions_abnormal,
+                    uniqIfState(session_id, errors > 0) as uniq_sessions_errored,
+                    uniqIfState(distinct_id, status == 2) as uniq_users_crashed,
+                    uniqIfState(distinct_id, status == 3) as uniq_users_abnormal,
+                    uniqIfState(distinct_id, errors > 0) as uniq_users_errored
+                FROM
+                    %(source_table_name)s
+                WHERE
+                    deleted == 0
+                GROUP BY
+                    org_id, project_id, started, release, environment
```

Comment:
You need yo group by toStartOfHour(timestamp) otherwise we would be aggregating per second in storage.

Reply:
It's an alias. changing to `hour_started`.